reana_client/cli/common.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA client cli functionality shared by multiple cli commands."""
+
+import click
+import logging
+import os
+import traceback
+
+
+def get_current_click_command(ctx):
+    """Return command (and subcommands) that were used to create given context.
+
+    For example, a context that was created by a command
+    '<executable name> code upload', would return 'code.upload'.
+
+    """
+    command = []
+    while ctx.parent:
+        command.append(ctx.info_name)
+        ctx = ctx.parent
+    return '.'.join(str(x) for x in reversed(command))
+
+
+def upload_files_to_server(user, organization, workflow, filenames, client):
+    """Upload files to REANA-Server.
+
+    Shared e.g. by `code upload` and `inputs upload`.
+
+    :param user: User ID
+    :param organization: Organization ID
+    :param workflow: ID of that Workflow whose workspace should be
+        used to store the files.
+    :param filenames: Absolute filepath(s) of files to be uploaded.
+    :param client: Instance of Bravado Client.
+    """
+    if workflow:
```

Comment:
I think this https://github.com/reanahub/reana-client/pull/79#discussion_r171538225 regarding throwing an exception still applies here, just to make the code more readable 😉